src/codegen/for-statement.ts

```diff
@@ -0,0 +1,47 @@
+import llvm from 'llvm-node';
+import ts from 'typescript';
+
+import LLVMCodeGen from '.';
+
+export default class CodeGenFor {
+  private cgen: LLVMCodeGen;
+
+  constructor(cgen: LLVMCodeGen) {
+    this.cgen = cgen;
+  }
+
+  public genForStatement(node: ts.ForStatement): void {
+    if (node.initializer) {
```

Comment:
```suggestion
    if (node.initializer && ts.isVariableDeclarationList(node.initializer))
```

Avoid nesting.

Reply:
node.initializer could be an expression. Although I have not added this branch judgment, it should be added later might.